smac/epm/gaussian_process.py

```diff
@@ -262,15 +264,30 @@ def _predict(self, X_test: np.ndarray, full_cov: bool = False) -> typing.Tuple[n
             raise Exception('Model has to be trained first!')
 
         X_test = self._impute_inactive(X_test)
-        mu, var = self.gp.predict(X_test, return_cov=True)
-        var = np.diag(var)
 
-        # Clip negative variances and set them to the smallest
-        # positive float value
-        var = np.clip(var, VERY_SMALL_NUMBER, np.inf)
+        predict_kwargs = {'return_cov': False, 'return_std': True}
+        if cov_return_type == 'full_cov':
+            predict_kwargs = {'return_cov': True, 'return_std': False}
 
-        if self.normalize_y:
-            mu, var = self._untransform_y(mu, var)
+        if cov_return_type is None:
+            mu = self.gp.predict(X_test)
+            var = None
+
+            if self.normalize_y:
+                mu = self._untransform_y(mu)
+
+        else:
+            mu, var = self.gp.predict(X_test, **predict_kwargs)
+
+            if cov_return_type == 'diagonal_cov':
+                var = var ** 2  # since we get standard deviation for faster computation
+
+            # Clip negative variances and set them to the smallest
+            # positive float value
+            var = np.clip(var, VERY_SMALL_NUMBER, np.inf)
+
+            if self.normalize_y:
+                mu, var = self._untransform_y(mu, var)
```

Comment:
Hm, I don't think this is correct if we want to return the standard deviation, as it assumes that this is the variance that needs un-normalization. Could you please have a look here?